import-export-cli/cmd/deployAPI.go

```diff
@@ -0,0 +1,113 @@
+` + utils.ProjectName + ` ` + DeployCmdLiteral + ` ` + DeployAPICmdLiteral + ` -n FacebookAPI -v 2.1.0 --rev 2 -r admin -g Label1 -e production --hide-on-devportal
+NOTE: All the 5 flags (--name (-n), --version (-v) , --rev and --gateway (-g) --environment (-e)) are mandatory.`
+
+// DeployAPICmd represents the deploy API command
+var DeployAPICmd = &cobra.Command{
+	Use: DeployAPICmdLiteral + " (--name <name-of-the-api> --version <version-of-the-api> --provider <provider-of-the-api> " +
+		"--rev <revision_number> --gateway <gateway-environment> " +
+		"--environment <environment-from-which-the-api-should-be-deployed>)",
+	Short:   deployAPICmdShortDesc,
+	Long:    deployAPICmdLongDesc,
+	Example: deployAPICmdExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + DeployAPICmdLiteral + " called")
+		cred, err := GetCredentials(deployCmdEnvironment)
+		if err != nil {
+			utils.HandleErrorAndExit("Error getting credentials", err)
+		}
+		gateways := impl.GenerateGatewayArray(args, deployCmdGateway, deployCmdHideOnDevportal)
+		executeDeployAPICmd(cred, gateways)
+	},
+}
+
+func executeDeployAPICmd(credential credentials.Credential, deployments []utils.Deployment) {
+	accessToken, preCommandErr := credentials.GetOAuthAccessToken(credential, deployCmdEnvironment)
+	if preCommandErr == nil {
+		resp, err := impl.ReflectDeploymentChangeInGatewayEnv("deploy-revision", accessToken, deployCmdEnvironment,
+			deployCmdAPIName, deployCmdAPIVersion, deployCmdProvider, deployCmdRevisionNum, deployments, false)
+		if err != nil {
+			utils.HandleErrorAndExit("Error while deploying the API", err)
+		}
+		// Print info on response
+		utils.Logf(utils.LogPrefixInfo+"ResponseStatus: %v\n", resp.Status())
+		if resp.StatusCode() == http.StatusCreated {
+			fmt.Println(apiNameForStateChange + " API state changed successfully!")
+		} else {
+			fmt.Println("Error while deploying the API: ", resp.Status(), "\n", string(resp.Body()))
+		}
+	} else {
+		fmt.Println("Error getting OAuth tokens to deploy the API:" + preCommandErr.Error())
+	}
+
+}
+
+// init using Cobra
+func init() {
+	DeployCmd.AddCommand(DeployAPICmd)
+	DeployAPICmd.Flags().StringVarP(&deployCmdAPIName, "name", "n", "",
+		"Name of the API to be deployed")
+	DeployAPICmd.Flags().StringVarP(&deployCmdAPIVersion, "version", "v", "",
+		"Version of the API to be deployed")
+	DeployAPICmd.Flags().StringVarP(&deployCmdProvider, "provider", "r", "",
+		"Provider of the API")
+	DeployAPICmd.Flags().StringVarP(&deployCmdGateway, "gateway", "g", "",
+		"Gateways which the revision has to be deployed")
+	DeployAPICmd.Flags().StringVarP(&deployCmdRevisionNum, "rev", "", "",
+		"Revision number of the API to be deployed")
+	DeployAPICmd.Flags().BoolVar(&deployCmdHideOnDevportal, "hide-on-devportal", false,
+		"Hide the gateway environment on devportal")
```

Comment:
```suggestion
		"Hide the gateway environment on the Developer Portal")
```

Reply:
Shall we change devportal to Developer Portal? It is better if we can use the full term (Developer Portal) in the explanations and documentation as recommended by the docs team.